base_layer/core/src/base_node/states/block_sync.rs

```diff
@@ -682,13 +682,7 @@ async fn request_headers<B: BlockchainBackend + 'static>(
                     "Failed to fetch header from peer: {:?}. Retrying.",
                     CommsInterfaceError::RequestTimedOut,
                 );
-                ban_sync_peer_if_online(
-                    shared,
-                    sync_peers,
-                    sync_peer.clone(),
-                    shared.config.block_sync_config.short_term_peer_ban_duration,
-                )
-                .await?;
+                exclude_sync_peer_if_online(shared, sync_peers, sync_peer.clone()).await?;
```

Comment:
Previously peers were excluded and not banned, this was changed to a short term ban as a bad node can be at the tip and never respond to block requests. This attack will cause the local node to continually switch between listening state and block sync attempting to download from the same bad node with the strongest tip resulting in the local node getting stuck without progressing to the second strongest tip.

Reply:
See #1829

Reply:
Understood, so I suppose a solution could be to have all ping peers included and ordered from strongest tip to least strongest tip where their tip > our tip, so that when the peer is excluded the equal or next strongest tip could be used. 

The main things to consider when banning are:
1. We cannot (non-trivial) automagically reconnect to that peer after the ban expires 
1. The banned peer will mark this node as offline (because it cannot connect to it for 5 minutes) and so won't connect to it again
1. False positives, if the internet connection suddenly drops/is delayed we could mistakenly ban a peer

Will think about this further, but for now I'll readd the banning. The connectivity manager may need to retry some offline peers if it has no more to use.

Reply:
I agree, I think the banning was a short term solution until a more comprehensive banning system can be implemented that tracks bad behaviour and enforce local punishment. Not responding to a few requests in a short period of time should not result in a ban, but if this is repeated behaviour where a node responds to some requests and not to others then a local ban might be warranted. The block sync state requires all sync peers to have exactly the same chain to function correctly, a transition from block sync to listening state is required when selecting a different tip and new sync peers to sync to.